qadence/noise/protocols.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+import importlib
+from dataclasses import dataclass
+from typing import Callable, Counter, cast
+
+PROTOCOL_TO_MODULE = {
+    "readout": "qadence.noise.readout",
+}
+
+
+@dataclass
+class Noise:
+    READOUT = "readout"
+
+    def __init__(self, protocol: str, options: dict = dict()) -> None:
+        self.protocol: str = protocol
```

Comment:
should we turn `protocol` into a `StrEnum`?

Reply:
Hm, I don't think we'd be changing this in this PR. Is this correct @Roland-djee?

Reply:
Probably not in this one. But yes I think that would be useful. The only drawback is that users would need to import another type, here it is all encapsulated in `Noise`.